### PR TITLE
rewrote as parameterized tests

### PR DESCRIPTION
**Pull request by asapekia**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

File: src/test/java/com/thealgorithms/searches/RabinKarpAlgorithmTest.java [hunk 1]
```diff
@@ -2,55 +2,26 @@
...
-    }
-
-    @Test
-    // valid test case
-    public void RabinKarpAlgorithmTestNotFound() {
-        String txt = "ABCBCBCAAB";
-        String pat = "AADB";
-        int value = RKA.search(pat, txt, q);
-        assertEquals(value, -1);
+  RabinKarpAlgorithm RKA = new RabinKarpAlgorithm();
```
Comment (vil02) [inline src/test/java/com/thealgorithms/searches/RabinKarpAlgorithmTest.java:9]: The `search` method is `static`. The `RKA` object is not needed.

File: src/test/java/com/thealgorithms/searches/RabinKarpAlgorithmTest.java [hunk 2]
```diff
@@ -2,55 +2,26 @@
...
+  int q = 101;
```
Comment (vil02) [inline src/test/java/com/thealgorithms/searches/RabinKarpAlgorithmTest.java:10]: What about testing for different values of `q`?

**Review by vil02 — CHANGES_REQUESTED** [review]
Could you please add a test case, which covers the situation in which the condition in the line:
https://github.com/TheAlgorithms/Java/blob/e5d33f35651e8b46ba2318dbd934dc3eeb81507b/src/main/java/com/thealgorithms/searches/RabinKarpAlgorithm.java#L60
Is satisfied?

File: src/test/java/com/thealgorithms/searches/RabinKarpAlgorithmTest.java (hunk 1 above)
Comment (asapekia) [reply src/test/java/com/thealgorithms/searches/RabinKarpAlgorithmTest.java:9]: It is not static.

Comment (vil02) [reply src/test/java/com/thealgorithms/searches/RabinKarpAlgorithmTest.java:9]: My bad. Sorry for the mistake - It is not. But it should be. However it is out of scope for this PR.

File: src/test/java/com/thealgorithms/searches/RabinKarpAlgorithmTest.java (hunk 2 above)
Comment (asapekia) [reply src/test/java/com/thealgorithms/searches/RabinKarpAlgorithmTest.java]: added testcases with different prime numbers(q).

**Comment by asapekia** [comment]
can someone help me with the formatting. I am unable to use clang formatter on my laptop.

**Comment by vil02** [comment]
> can someone help me with the formatting. I am unable to use clang formatter on my laptop.

Please have a look [here](https://github.com/TheAlgorithms/Java/pull/4438#issuecomment-1741399556).

I pushed 2276b72fd3d736633b94e7ffe3243ebad41c16b5 - it should fix the formatting.

**Comment by asapekia** [comment]
> Could you please add a test case, which covers the situation in which the condition in the line:
> 
> https://github.com/TheAlgorithms/Java/blob/e5d33f35651e8b46ba2318dbd934dc3eeb81507b/src/main/java/com/thealgorithms/searches/RabinKarpAlgorithm.java#L60
> 
> 
> Is satisfied?
> And please fix the formatting.

yes I have added the test cases that will produce negative hash(t).

**Comment by asapekia** [comment]
> > can someone help me with the formatting. I am unable to use clang formatter on my laptop.
> 
> Please have a look [here](https://github.com/TheAlgorithms/Java/pull/4438#issuecomment-1741399556).
> 
> I pushed [2276b72](https://github.com/TheAlgorithms/Java/commit/2276b72fd3d736633b94e7ffe3243ebad41c16b5) - it should fix the formatting.

Thanks @vil02. This PR can be merged now.

**Review by vil02 — APPROVED** [review]
@asapekia would you like to _clean-up_ `RabinKarpAlgorithm`?

**Comment by asapekia** [comment]
> @asapekia would you like to _clean-up_ `RabinKarpAlgorithm`?

yes. Working on it.

**Pull request merged**